File: claude_code_is_programmable_2.py
#!/usr/bin/env -S uv run --script

import subprocess

prompt = """

GIT checkout a NEW branch.

CREATE ./cc_todo/todo.ts: a zero library CLI todo app with basic CRUD. 

THEN GIT stage, commit and SWITCH back to main.

"""

command = ["claude", "-p", prompt, "--allowedTools", "Edit", "Bash", "Write"]

process = subprocess.run(command, check=True)

print(f"Claude process exited with output: {process.stdout}")

File: claude_code_is_programmable_2.js
import { spawn } from 'child_process';

const prompt = `git checkout a NEW branch. 
CREATE ./cc_todo/todo.ts: a zero library CLI todo app with basic CRUD. 
THEN git stage, commit and SWITCH back to main.`;

const command = 'claude';
const args = ['-p', prompt, '--allowedTools', 'Edit', 'Bash', 'Write'];

const child = spawn(command, args, {
  stdio: 'inherit'
});

child.on('close', (code) => {
  console.log(`Claude process exited with code ${code}`);
});